src/code/z_fbdemo_circle.c

```diff
@@ -1,19 +1,142 @@
+    gsSPEndDisplayList(),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_circle/TransitionCircle_Init.s")
+//! @bug: TransitionCircle_Update should take an additional argument `s32 updateRate`
+const TransitionInit TransitionCircle_InitVars = {
+    TransitionCircle_Init,   TransitionCircle_Destroy, (void*)TransitionCircle_Update, TransitionCircle_Draw,
+    TransitionCircle_Start,  TransitionCircle_SetType, TransitionCircle_SetColor,      NULL,
+    TransitionCircle_IsDone,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_circle/TransitionCircle_Destroy.s")
+void TransitionCircle_Start(void* thisx) {
+    TransitionCircle* this = (TransitionCircle*)thisx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_circle/TransitionCircle_Update.s")
+    this->stepValue = 0.1f;
+    if (this->direction == 0) {
+        this->unk_10 = 0.0f;
+        this->unk_0C = 1.0f;
+    } else {
+        this->unk_0C = 0.0f;
+        this->unk_10 = 1.0f;
+    }
+    this->unk_04 = this->unk_0C;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_circle/TransitionCircle_SetColor.s")
+void* TransitionCircle_Init(void* thisx) {
+    TransitionCircle* this = (TransitionCircle*)thisx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_circle/TransitionCircle_SetType.s")
+    bzero(this, sizeof(TransitionCircle));
+    this->unk_15 = 1;
+    this->texture = gCircleTex;
+    this->unk_1C = 6;
+    this->unk_1D = 6;
+    this->unk_1E = 4;
+    this->unk_1F = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_circle/func_80164C14.s")
+    return this;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_circle/TransitionCircle_Draw.s")
+void TransitionCircle_Destroy(void* thisx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_circle/TransitionCircle_IsDone.s")
+void TransitionCircle_Update(void* thisx) {
+    TransitionCircle* this = (TransitionCircle*)thisx;
+
+    this->isDone = Math_StepToF(&this->unk_04, this->unk_10, this->stepValue);
+}
+
+void TransitionCircle_SetColor(void* thisx, u32 color) {
+    TransitionCircle* this = (TransitionCircle*)thisx;
+
+    this->color.rgba = color;
+}
+
+void TransitionCircle_SetType(void* thisx, s32 type) {
+    TransitionCircle* this = (TransitionCircle*)thisx;
+
+    if (type & TC_SET_PARAMS) {
+        this->unk_15 = type & 1;
+    } else if (type == 1) {
+        this->direction = 1;
+    } else {
+        this->direction = 0;
+    }
+}
+
+void TransitionCircle_LoadAndSetTexture(Gfx** gfxp, TexturePtr texture, s32 fmt, s32 arg3, s32 arg4, s32 arg5,
+                                        f32 arg6) {
+    Gfx* gfx = *gfxp;
+    s32 xh = D_801FBBCC;
+    s32 yh = D_801FBBCE;
+    s32 width = 1 << arg4;
+    s32 height = 1 << arg5;
+    f32 s;
+    f32 t;
+    s32 dtdy;
+    s32 dsdx;
+
+    gDPLoadTextureBlock_4b(gfx++, texture, fmt, width, height, 0, G_TX_MIRROR | G_TX_CLAMP, G_TX_MIRROR | G_TX_CLAMP,
+                           arg4, arg5, G_TX_NOLOD, G_TX_NOLOD);
```

Comment:
```suggestion
    gDPLoadTextureBlock_4b(gfx++, texture, fmt, width, height, 0, G_TX_MIRROR | G_TX_CLAMP, G_TX_MIRROR | G_TX_CLAMP,
                           masks, maskt, G_TX_NOLOD, G_TX_NOLOD);
```
Can name these from the macro as well.